serverlessworkflow/modules/ROOT/pages/core/understanding-workflow-error-handling.adoc

```diff
@@ -1 +1,77 @@
-//= Understanding workflow error handling
\ No newline at end of file
+= Understanding workflow error handling
+
+This guide explains how to handle errors that might occur while executing workflow states. Although a non desirable situation, Workflow states, as part of their interaction with external systems, might fail. For these situations, Serverless Workflow specification provides an link:{spec_doc_url}#workflow-error-handling[error handling] mechanism which allows users to deal with these rainy day circumstances.
+
+An error is an exceptional occurrence that alters the regular Workflow execution sequence. When an error occurs, rather than transitioning to the state that would have been executed in case of a successful execution, the flow, if configured to do so, jumps instead to an alternative state, potentially capable of dealing with it. Note that error definition is always optional, if no error handling is defined, the Workflow execution will be aborted when something goes wrong. If you are a software developer, you can can think of Workflow error handling  as a `try-catch` or a `goto` construct.
+
+== Error definition
+
+An link:{spec_doc_url}#error-definition[Error definition] in Serverless Workflow is composed of a name and an code. The name is a natural language description of the error, for example: “wrong parameter”, while the code is a vendor specific parameter that helps the implementor to identify the error. In the case of Kogito, the code is mandatory and consist of the fully qualified Java class name of an link:https://docs.oracle.com/javase/tutorial/essential/exceptions/index.html[Exception type] that might be thrown by the state.
+
+The algorithm used by Kogito to determine if an exception should be handled by an error definition is:
+
+* it checks if the type of the Java Exception being thrown is an instance (which includes any descendant) of the Java Exception specified in `code` property of the error definition. 
+* if this not the case, it recursively checks if the root cause of the Java Exception is an instance (which includes any descendant) of the Java Exception specified in `code` property of the error definition.
```

Comment:
checks if the root cause.... is duplicated from the first bullet, maybe rephrase it?

Reply:
yes, I normally dislike duplication, but in this case,  rephrasing the second bullet might indicate the procedure is different for the root cause, which is not. 
What about "if this not the case, it recursively performs the previous checks over the the chains of root causes of the Java Exception"?

Reply:
changed, please take a look @spolti